src/components/CommandPalette.tsx

```diff
@@ -210,17 +209,30 @@ const CommandRow: FC<{
                         )
                       })()
                     : // For other commands, use normal highlighting
-                      gestureInProgress.length
+                      shortcut.id === 'cancel' && selected
+                      ? 1 // Highlight the cancel gesture when any gesture is in progress
+                      : shortcut.id === 'cancel'
+                        ? undefined
+                        : gestureInProgress.length
```

Comment:
This would be better expressed as:

```ts
    shortcut.id === 'cancel' 
      ? (selected ? 1 : undefined)
      : gestureInProgress.length
```

That way the cancel logic is consolidated and you avoid the duplication of `shortcut.id === 'cancel'`.